core/src/value/string.rs

```diff
@@ -37,6 +42,66 @@ impl<'js> String<'js> {
     }
 }
 
+/// Rust representation of a JavaScript C string.
+#[derive(Debug)]
+pub struct CString<'js> {
+    ptr: *const c_char,
```

Comment:
This should probably be a `NonNull`.
Feels more correct as this pointer should never be null and might allow for more optimizations within enums.

Reply:
Agreed